lib/src/services/databases/dataRefresh/implementation.dart

```diff
@@ -0,0 +1,47 @@
+import "package:ramaz/models.dart";
+import "package:ramaz/services.dart";
+import "../../firestore.dart";
+import "interface.dart";
+///
+class CloudDataRefresh implements DataRefreshInterface{
+  ///
+  static final CollectionReference<Map> data = Firestore
+      .instance.collection("dataRefresh");
+
+  @override
+  Future<void> getRefreshData() async{
+    final Map<String, String> map = Map<String,String>.from(await data
+        .doc("dataRefresh").throwIfNull("Cannot get refreshed data"));
+    if(DateTime.parse(map["user"]!).isAfter(Services.instance.prefs
+        .getRefreshData("user")!)||
+        Services.instance.prefs.getRefreshData("user")==null){
+        await Services.instance.database.user.signIn();
+        await Models.instance.user.init();
+        print(map["user"]);
+    }
+    if(DateTime.parse(map["schedule"]!).isAfter(Services.instance.prefs
+        .getRefreshData("schedule")!)){
+          await Models.instance.schedule.init();
+          print(map["schedule"]);
+    }
```

Comment:
1. I noticed the condition isn't exactly the same between these two -- is that important?
2. You made `SharedPreferences.getRefreshData` return `DateTime?`, but you immediately use `!`. Maybe you can make it non-nullable with a helpful error message if the right key can't be found. 
3. The keys passed to `SharedPreferences` are simple nouns like `schedule` and `user`. You can imagine some future confusion because it sounds like it actually holds the `user`/`schedule` _data_. Maybe clarify by making it `last_update_schedule` or `user_last_update`. 
4. You don't want to sign in any services here, because you should already be fully signed in. You can't access Firestore unless you're already authenticated, so you can't possibly have gotten the data above and not be signed in yet. 
4. You don't want to be touching any models here. Services are fully initialized before the models are even touched, and that's by design. Each `XModel` should get its data from a `HybridX`. If you want to change the data that the model will get, make sure to change it in the database, and the model will pick it up. That's also why the cloud and local databases are separated but collected together in the `HybridDatabase` -- so that the model can use the local data when appropriate, and assume that it's in-sync with the cloud data. 
5. The repetitive structure and the deep coupling (ie, that `CloudDataRefresh` knows about `HybridUser` and `HybridCalendar` and depends on them to exist) show that you need to put this all somewhere else. See my above suggestions.

Reply:
but for the User Model isnt .signIn where it actually downloads the data?

Reply:
Oh you're right, I was confusing that with the idea of actually authenticating (maybe that means we need to rename the function, maybe like `onSignIn` to show it only happens _after_ sign-in, or `saveData` to show that's all we're doing). 

So yes, `signIn` would be the right function to call, but again, not from here.

Reply:
> I noticed the condition isn't exactly the same between these two -- is that important?

I just forgot to add it to the other one oops

Reply:
> So yes, signIn would be the right function to call, but again, not from here.

Why not from here?

Reply:
> You don't want to be touching any models here. Services are fully initialized before the models are even touched, and that's by design. Each XModel should get its data from a HybridX. If you want to change the data that the model will get, make sure to change it in the database, and the model will pick it up. That's also why the cloud and local databases are separated but collected together in the HybridDatabase -- so that the model can use the local data when appropriate, and assume that it's in-sync with the cloud data.

Also don't I need to reinitialize the model the properly store the data in the right places to be presented in the UI??

Reply:
> Why not from here?

Because you should be keeping the services separate, and this service is doing way too much work in other services. You're better off building this functionality directly into the other services so they can stay independent. 

> Also don't I need to reinitialize the model the properly store the data in the right places to be presented in the UI??

You don't _re_-initialize, because they haven't been initialized. The models assume that `HybridX.getY()` will always give the right data, and `HybridX.getY` assumes that the local database is up-to-date. Since the last assumption is wrong, all you need to do is just fill the local database with the updated cloud data and let RamLife handle the rest.